symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/ServiceLookup.java

```diff
@@ -18,19 +18,20 @@
    * Load a service implementation class using {@link ServiceLoader}.
    *
    * @return a service implementation class.
+   * @throws IllegalStateException if no implementation or several implementations found in classpath.
    */
   public static <T> T lookupSingleService(Class<T> clz) {
 
     final ServiceLoader<T> classServiceLoader = ServiceLoader.load(clz);
 
     final List<T> services = StreamSupport.stream(classServiceLoader.spliterator(), false)
-            .collect(Collectors.toList());
+        .collect(Collectors.toList());
 
     if (services.isEmpty()) {
       throw new IllegalStateException("No service implementation found in classpath.");
```

Comment:
one issue I had yesterday when trying the BDK on a new project was that I only added the bdk-core dependencies so I had the _No service implementation found in classpath._ error message 
which was not very helpful

At least it could also tell which implem it was looking for that would have helped me to figure out that an HTTP client impl was missing

And even better the error message could help to understand why the issue happens in the first place:
- suggesting that I add the missing impl dependency
- suggesting that I remove one of them (and listing the ones that are present would help too)